Changelog for mlflow 1.7

mlflow-automation opened the pull request:
2020-03-01-22-50-57-702399 Update

Commit:
Generating changelog for MLflow 1.7.0

Commit:
change log for 1.7 release

dbczumar commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,38 @@
 Changelog
 =========
+1.7.0 (2020-03-01)
```
@mparkhe I don't see entries here for the `mlflow gc` command. Were cherry-picked features not included in this autogenerated change log?

cc @smurching

smurching replied:
Ah, we need to add labels to PRs between the 1.6 & 1.7 releases for them to show up in the changelog (will update the automation to throw when there are unlabeled PRs, & also go back and label the old PRs)

smurching replied:
Ideally we could also enforce that a PR has a release-notes label before merging

mparkhe replied:
yeah! i added labels to most of the PRs -- i must have forgotten the original gc PR

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,38 @@
+MLflow 1.7.0 includes several major features and improvements:
+
+Model Registry APIs: breaking changes
+- REST APIs are no longer in beta and intended to be stable until the next major version. As such, this release includes a number of breaking changes.
```
It would be nice to mention why they changed, i.e. to clean up the API or make it more consistent with other APIs.

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,38 @@
+------------------
+MLflow 1.7.0 includes several major features and improvements:
+
+Model Registry APIs: breaking changes
```
This is only REST APIs, right, not Python APIs? Most users won't see the REST ones at all so it would be good to say REST APIs. Also say what the impact is (which clients are compatible with new/old servers: do you have to upgrade both? if not, how long is compatibility going to be supported for?)

mparkhe replied:
Thanks @mateiz. Made the title more clear and added comments about clients maintaining backward compatibility at the API level and the user code does not need to change.

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,38 @@
+
+Other Features:
+
+- Fix bar chart bug introduced in #2393 (#2498, @smurching)
```
People don't know what #2393 is so say which MLflow release this was in (or even better, which bug)

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,38 @@
+
+- Fix bar chart bug introduced in #2393 (#2498, @smurching)
+- Save metric plot state in URLs for easier link sharing (#2393, @smurching)
+- Update R client to call record logged model  (#2430, @tomasatdatabricks)
```
Isn't this part of the "add log model endpoint"? We should group these into topics, instead of having a separate line for each PR.

mateiz commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,38 @@
+- Save metric plot state in URLs for easier link sharing (#2393, @smurching)
+- Update R client to call record logged model  (#2430, @tomasatdatabricks)
+- Fix handleLayoutChange to update dragmode (#2459, @harupy)
+- Announce deprecation of Python 2 support in MLflow (#2438, @smurching)
```
This should say "this release deprecates Python 2", and it's major enough that it should be part of the major changes.

smurching committed:
Update CHANGELOG.rst

smurching committed:
Update CHANGELOG.rst

smurching commented:
Addressed some of the comments (& added labels to missing PRs & regenerated the changelog), will let @mparkhe handle updates to the model registry portions of the changelog - thanks @mparkhe @mateiz @dbczumar for generating the changelog/reviewing!

Commit:
updating change logs

- moved Python 2 deprecation at the top
- Section for Model Registry REST API breaking change

mparkhe commented:
Thanks @smurching. I moved the Python 2 deprecation notice up top and updated Model Registry API changes section to be more human readable. Non-breaking changes for Model Registry (UI updates, client updates -- moved below)

Commit:
formatting

sueann commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,50 @@
+release Model Registry APIs are no longer beta and intended to be stable until the next major
+version. Python and Java client APIs for Model Registry are backward compatible and have been
+updated to use new REST APIs. However, when using MLflow client with a server using updated REST
+endpoints, you will need to upgrade to new client version without changing any user code.
```
also, we can refer to the relevant PRs here instead of listing them in "Other Features" (now they are doubly mentioned)

mparkhe replied:
I will move the API related stuff (UI and client you marked below)

sueann commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,50 @@
+- MLflow UI page titles now reflect the content displayed on the page (#2420, @AveshCSingh)
+- Adds a new ``LogModel`` REST API endpoint for capturing model metadata, and call it from the Python and R clients (#2369, #2430, #2468 @tomasatdatabricks)
+- The Model Registry UI has been updated to use new REST APIs (#2476 @aarondav; #2507, @mparkhe)
+- Updated MLflow Python and Java clients to use new REST APIs. The client APIs have minor changes and deprecated arguments however for this release are backward compatible. These deprecated arguments will be dropped in future releases. (#2457, @tomasatdatabricks; #2502, @mparkhe)
```
merge these into the breaking changes blurb above

stbof commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,50 @@
+release Model Registry APIs are no longer beta and intended to be stable until the next major
+version. Python and Java client APIs for Model Registry are backward compatible and have been
+updated to use new REST APIs. However, when using MLflow client with a server using updated REST
+endpoints, you will need to upgrade to new client version without changing any user code.
```
However, when using MLflow client with a server using updated REST
endpoints, you will need to upgrade to new client version without changing any user code.

-> However, when using an MLflow client with a server using updated REST
endpoints, you won't need to change any code but will need to upgrade to a new client version.

stbof commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,50 @@
+- Updated Model Registry documentation page with code snippets and examples (#2493, dmatrix)
+- Better error message for Model Registry, when using incompatible backend server (#2456, @aarondav)
+- matplotlib is no longer required to use XGBoost and LightGBM autologging (#2423, @harupy)
+- Fixes bug where matplotlib figures were not closed in XGBoost and LightGBM autologging (#2386, @harupy)
```
```suggestion
- Fix bug where matplotlib figures were not closed in XGBoost and LightGBM autologging (#2386, @harupy)
```

stbof commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,50 @@
+- Improve readability of run table column selector nodes (#2388, @dbczumar)
+- Validate experiment name supplied to ``UpdateExperiment`` REST API endpoint (#2357, @ggliem)
+- Fix broken MLflow DB README link in CLI docs (#2377, @dbczumar)
+- Change Copyright year across docs to 2020 (#2349, @ParseThis)
```
```suggestion
- Change copyright year across docs to 2020 (#2349, @ParseThis)
```

stbof commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,50 @@
+- Better error message for Model Registry, when using incompatible backend server (#2456, @aarondav)
+- matplotlib is no longer required to use XGBoost and LightGBM autologging (#2423, @harupy)
+- Fixes bug where matplotlib figures were not closed in XGBoost and LightGBM autologging (#2386, @harupy)
+- Fix parameter reading logic to support param values with newlines in FileStore (#2376, @dbczumar)
```
```suggestion
- Fixed parameter reading logic to support param values with newlines in FileStore (#2376, @dbczumar)
```

stbof commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,50 @@
+- Fix parameter reading logic to support param values with newlines in FileStore (#2376, @dbczumar)
+- Improve readability of run table column selector nodes (#2388, @dbczumar)
+- Validate experiment name supplied to ``UpdateExperiment`` REST API endpoint (#2357, @ggliem)
+- Fix broken MLflow DB README link in CLI docs (#2377, @dbczumar)
```
```suggestion
- Fixed broken MLflow DB README link in CLI docs (#2377, @dbczumar)
```

stbof commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,50 @@
+
+Other Features:
+
+- Adds ability to click through to individual runs from metrics plot (#2295, @harupy)
```
```suggestion
- Added ability to click through to individual runs from metrics plot (#2295, @harupy)
```

stbof replied:
Change Adds to Added throughout.

stbof commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,50 @@
+
+Breaking changes to Model Registry REST APIs:
+
+Model Registry REST APIs are updated to be more consistent with the other MLflow APIs. With this
```
```suggestion
Model Registry REST APIs have been updated to be more consistent with the other MLflow APIs. With this
```

Commit:
Apply suggestions from @stbof

Co-Authored-By: Stephanie Bodoff <stephanie.bodoff@databricks.com>

Commit:
addressing @sueann and @stbof 's comments

sueann commented on CHANGELOG.rst:
```diff
@@ -12,38 +12,38 @@ guide.
 
-- Adds ability to click through to individual runs from metrics plot (#2295, @harupy)
+- Added ability to click through to individual runs from metrics plot (#2295, @harupy)
 - Adds ``mlflow gc`` CLI for permanent deletion of runs (#2265, @t-henri)
```
"Added" here and below

Commit:
reformat

Commit:
`adds` -> `added`

sueann commented on CHANGELOG.rst:
```diff
@@ -1,5 +1,49 @@
+that you upgrade to Python 3 - see  https://docs.python.org/3/howto/pyporting.html for a migration
+guide.
+
+Breaking changes to Model Registry REST APIs:
```
i don't think we need this heading, but let's check this in for the release now - we can always tweak later.

sueann reviewed (approved):
lgtm
